zerver/data_import/slack.py

```diff
@@ -472,6 +472,14 @@ def build_avatar_url(slack_user_id: str, user: ZerverFieldsT) -> tuple[str, str]
         avatar_hash = user["profile"]["avatar_hash"]
         avatar_url = f"https://ca.slack-edge.com/{team_id}-{slack_user_id}-{avatar_hash}"
         avatar_source = UserProfile.AVATAR_FROM_USER
+    elif user.get("is_integration_bot"):
+        # Unlike other Slack user types, Slacks integration bot avatar URL is
+        # PNG image URL.
+        avatar_url = user["profile"]["image_72"]
+        if avatar_url.endswith((".png", ".jpg")):
```

Comment:
I think we don't care what the type is as long as we can process it; I wouldn't want this code to fail if Slack allows `webp` images there at some point. I think what we should do follow the same approach we use when uploading an avatar -- guess the content-type from the filename, and ask if we can thumbnail it. Here's the code there:

```
def upload_avatar_image(                                                                                
    user_file: IO[bytes],                                                                               
    user_profile: UserProfile,                                                                          
    content_type: str | None = None,                                                                    
    backend: ZulipUploadBackend | None = None,                                                          
    future: bool = True,                                                                                
) -> None:                                                                                              
    if content_type is None:                                                                            
        content_type = guess_type(user_file.name)[0]                                                    
    if content_type not in THUMBNAIL_ACCEPT_IMAGE_TYPES:                                                
        raise BadImageError(_("Invalid image format"))                                                  
```

Reply:
Adjusted `build_avatar_url` to be:

```python
def build_avatar_url(slack_user_id: str, user: ZerverFieldsT) -> tuple[str, str]:
    ...
    elif user.get("is_integration_bot"):
        # Unlike other Slack user types, Slacks integration bot avatar URL ends with
        # a file type extension (.png, in this case).
        # e.g https://avatars.slack-edge.com/2024-05-01/7218497908_deb94eac4c_512.png
        avatar_url = user["profile"]["image_72"]
        content_type = guess_type(avatar_url)[0]
        if content_type not in THUMBNAIL_ACCEPT_IMAGE_TYPES:
            logging.info(
                "Unsupported avatar type (%s) for user -> %s\n", content_type, user.get("name")
            )
            avatar_source = UserProfile.AVATAR_FROM_GRAVATAR
        else:
            avatar_source = UserProfile.AVATAR_FROM_USER
    else:
        logging.info("Failed to process avatar for user -> %s\n", user.get("name"))
    return avatar_source, avatar_url
```